### PR TITLE
Test for Layered Feature Map

### PR DESCRIPTION
**Pull request by David-Kreplin**
solves #133

### CODE REVIEW

**Comment by rothma** [comment]
I think adding tests for the executability of the final feature maps would be helpful (for both, from_string and regularly build ones).

**Comment by MoritzWillmann** [comment]
Also, it could make sense to use the parameterize fixture

**Comment by David-Kreplin** [comment]
> Also, it could make sense to use the parameterize fixture

What do you mean with parameterize fixture?

**Comment by rothma** [comment]
> > Also, it could make sense to use the parameterize fixture
> 
> What do you mean with parameterize fixture?

You can define resuable fixtures (i.e. instances of classes) that can be called in the tests without redefining it in every test. If you want one test to apply to multiple fixutres, you can parametrize them in a decorator. You can look at the [qsvc tests](https://github.com/sQUlearn/squlearn/blob/main/tests/kernel/ml/test_qsvc.py) as examples. Here, projected and fidelity instances are defined as fixtures which are then called with the parametrized "qsvc" variable, e.g. in test_predict (line 80)

File: tests/feature_map/test_layered_feature_map.py [hunk 1]
```diff
@@ -194,11 +213,37 @@ def add_AA(gate_function, p, offset=0):
...
-            assert str(lfm.get_circuit([], p)) == str(expected_circuit)
+            assert str(lfm.get_circuit(x, p)) == str(expected_circuit)
+
+        kernel = FidelityKernel(
+            lfm, Executor("statevector_simulator"), initial_parameters=0.5 * np.ones(72)
+        ).evaluate(np.array([[]]), np.array([[]]))
+        assert np.allclose(kernel, np.array([1.0]))
+
+        # Won't run because of a bug in qiskit
+        # See https://github.com/Qiskit/qiskit/issues/8723
```
Comment (MoritzWillmann) [inline tests/feature_map/test_layered_feature_map.py:228]: Then remove it? 😄

Comment (David-Kreplin) [reply tests/feature_map/test_layered_feature_map.py:228]: Well it is implemented an should work once the qiskit bug is fixed. I would like to keep it.

**Comment by David-Kreplin** [comment]
Added 

> > > Also, it could make sense to use the parameterize fixture
> > 
> > 
> > What do you mean with parameterize fixture?
> 
> You can define resuable fixtures (e.g. instances of classes) that can be called in the tests without redefining it in every test. If you want one test to apply to multiple fixutres, you can parametrize them in a decorator. You can look at the [qsvc tests](https://github.com/sQUlearn/squlearn/blob/main/tests/kernel/ml/test_qsvc.py) as examples. Here, projected and fidelity instances are defined as fixtures which are then called with the parametrized "qsvc" variable, e.g. in test_predict (line 80)

I understand! In this case, however, building the longer feature map is also kind of a test.

**Comment by David-Kreplin** [comment]
Pull request for #133

File: tests/feature_map/test_layered_feature_map.py (hunk 1 above)
Comment (MoritzWillmann) [reply tests/feature_map/test_layered_feature_map.py:228]: Alright

**Comment by MoritzWillmann** [comment]
> Added
> 
> > > > Also, it could make sense to use the parameterize fixture
> > > 
> > > 
> > > What do you mean with parameterize fixture?
> > 
> > 
> > You can define resuable fixtures (e.g. instances of classes) that can be called in the tests without redefining it in every test. If you want one test to apply to multiple fixutres, you can parametrize them in a decorator. You can look at the [qsvc tests](https://github.com/sQUlearn/squlearn/blob/main/tests/kernel/ml/test_qsvc.py) as examples. Here, projected and fidelity instances are defined as fixtures which are then called with the parametrized "qsvc" variable, e.g. in test_predict (line 80)
> 
> I understand! In this case, however, building the longer feature map is also kind of a test.

I meant for readability and reusability of the test but I'm fine with it, as it is now, aswell

**Review by MoritzWillmann — APPROVED** [review]

**Pull request merged**